idaes/gas_distribution/unit_models/pipeline.py

```diff
@@ -0,0 +1,394 @@
+            rugosity = 0.025 * pyunits.mm
+        rug_value = pyo_value(rugosity)
+        rug_units = pyunits.get_units(rugosity)
+        self.rugosity = Param(
+            initialize=rug_value,
+            units=rug_units,
+            #doc=TODO,
+            mutable=True,
+        )
+
+        diam_mm = pyunits.convert(self.diameter, pyunits.mm)
+        friction_factor_expr = (2*log10(3.7*diam_mm/self.rugosity))**-2
+
+        self.friction_factor = Var(
+            initialize=pyo_value(friction_factor_expr),
+            units=pyunits.dimensionless,
+            doc="This is fraction factor lambda_l in V. Zavala's paper",
+        )
+
+        self.friction_factor_eqn = Constraint(expr=(
+            self.friction_factor == friction_factor_expr
+        ))
+
+    def add_diameter(self):
+        # We make diameter a variable as it may be more convenient
+        # to fix diameter than to fix area.
+        diam_units = pyunits.m
+        self.diameter = Var(
+            initialize=1.0,
+            units=diam_units,
+            doc="Diameter of the pipeline",
+        )
+
+        area = self.control_volume.area
+        def diameter_rule(b):
+            return (
+                pyunits.convert(area, diam_units**2) 
+                == Constants.pi * self.diameter**2 / 4.0
+            )
+        self.diameter_eqn = Constraint(
+            rule=diameter_rule,
+            doc="Equation linking control volume area and pipeline diameter",
+        )
+
+    #def add_speed_of_sound(self):
+    #    # FIXME: Attaching this property to the unit model is temporary.
+    #    # See comment above in the build method.
+    #    time = self.flowsheet().time
+    #    space = self.control_volume.length_domain
+    #    self.compressibility = Param(
+    #        initialize=0.80,
+    #        units=pyunits.dimensionless,
+    #        doc="Gas compressibility",
+    #    )
+    #    def speed_sound_rule(b, t, x):
+    #        state = self.control_volume.properties[t, x]
+    #        # FIXME: Embedded constant: "Vap"
+    #        gamma = state.heat_capacity_ratio_phase["Vap"]
+    #        z = self.compressibility
+    #        return (
+    #            pyunits.convert(sqrt(
+    #                    gamma * z * Constants.gas_constant * state.temperature
+    #                    / state.mw
+    #                ),
+    #                #pyunits.m**2/pyunits.s**2,
+    #                pyunits.m/pyunits.s,
+    #            )
+    #        )
+    #    # TODO: This expression has a very long and incomprehensible string
+    #    # representation due to the Shomate equations used in the heat
+    #    # capacity ratio. Can I make this string representation more
+    #    # comprehensible without adding variables for heat capacities?
+    #    self.speed_of_sound = Expression(
+    #        time,
+    #        space,
+    #        rule=speed_sound_rule,
+    #        doc="Speed of sound in the pipeline gas",
+    #    )
```

Comment:
Can this be removed?

Reply:
Thanks for pointing this out, yes looks like I forgot to remove this.